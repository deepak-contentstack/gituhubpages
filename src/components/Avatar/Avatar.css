@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';

.AvatarList {
	align-items: center;
	display: flex;
	justify-content: flex-start;

	&__expand-button {
		align-items: center;
		background-color: $color-base-gray-5;
		border: $space-2 solid $color-base-white-5;
		border-radius: $radii-50;
		color: $color-base-gray-15;
		display: flex;
		font-size: $size-font-tiny;
		font-weight: $font-weight-semi-bold;
		height: 1.65rem;
		justify-content: center;
		padding: 0;
		width: 1.65rem;

		&--tiny-text {
			font-size: 0.5rem;
		}
	}

	&__expand-link {
		color: $color-base-purple-5;
		font-size: $size-font-small;
		font-weight: $font-weight-bold;
		margin-left: $space-5;
		text-decoration: none;

		&--dark {
			color: $color-base-gray-15;
		}
	}

	& > :nth-child(2) {
		transform: translateX(-5px);
	}

	& > :nth-child(3) {
		transform: translateX(-10px);
	}

	& > :nth-child(4) {
		transform: translateX(-15px);
	}

	& > :nth-child(5) {
		transform: translateX(-20px);
	}

	& > :nth-child(6) {
		transform: translateX(-25px);
	}

	& > :nth-child(7) {
		transform: translateX(-30px);
	}

	& > :nth-child(8) {
		transform: translateX(-35px);
	}

	& > :nth-child(9) {
		transform: translateX(-40px);
	}

	& > :nth-child(10) {
		transform: translateX(-45px);
	}
}

.Avatar {
	align-items: center;
	background-color: $color-base-gray-5;
	border: $space-2 solid $color-base-white-5;
	border-radius: $radii-50;
	display: flex;
	font-family: $font-family-primary;
	font-size: 0.625rem;
	font-weight: $font-weight-extra-bold;
	justify-content: center;
	position: relative;

	&--large{
		height: 2rem;
		width: 2rem;
	}

	&--small{
		height: 1.625rem;
		width: 1.625rem;
		.Avatar__image {
			height: $space-24;
			width: $space-24;
		}
		.Avatar__active{
			border: 1px solid $color-base-white-5;
			bottom: 0;
			height: $space-4;
			right: 0;
			width: $space-4;
		}
	}

	&__link {
		align-items: center;
		color: $color-base-white-5;
		cursor: pointer;
		display: flex;
		height: 100%;
		justify-content: center;
		overflow: hidden;
		text-decoration: none;
		text-transform: uppercase;
		transition: background-color 300ms ease, color 300ms ease;
		width: 100%;
		&__grey{
			color:#647696;
		}
    &__white{
      color: #ffffff;
    }
	}

	&__active{
		background: $color-brand-success-base;
		border: $space-2 solid $color-base-white-5;
		border-radius: $radii-50;
		bottom: -$space-2;
		box-sizing: content-box;
		height: $space-8;
		position: absolute;
		right: -$space-2;
		width: $space-8;
	}

	&--single {
		border: none;
		font-size: $size-font-tiny;
		height: 2rem;
		width: 2rem;

		.Avatar__image {
			filter: grayscale(1);
			transition: filter 300ms ease;

			&:hover {
				filter: grayscale(0);
			}
		}
	}

	&__image {
		border-radius: $radii-50;
		height: auto;
		left: 50%;
		min-width: 100%;
		position: absolute;
		top: 50%;
		transform: translate(-50%, -50%);
		width: auto;
	}

	&__name {
		color: $color-base-gray-20;
		font-size: $size-font-small;
		font-weight: $font-weight-bold;
		margin: 0 0 $space-10 0;
	}

	&__email {
		color: $color-base-gray-20;
		display: block;
		font-size: $size-font-small;
		font-weight: $font-weight-medium;
	}

	&__profile {
		align-items: center;
		display: flex;
		justify-content: flex-start;

		&:hover {
			.Avatar__name {
				color: $color-base-purple-5;
			}
		}

		.Avatar__image {
			border: $space-2 solid $color-base-white-5;
			border-radius: $radii-50;
			height: auto;
			margin-right: $space-10;
			min-width: 0;
			position: static;
			transform: none;
			width: 1.625rem;
		}

		.Avatar__name {
			color: $color-base-black-5;
			font-size: $size-font-medium;
			font-weight: $font-weight-regular;
			margin: 0;
		}
	}

	&__dropdown {
		z-index: 999;
		.Dropdown__menu__list__item {
			padding: $space-4 $space-10;
		}
	}
}

/* Pop up CSS */
.popUp-body__header {
    align-items: center;
    display: flex;
    font-size: $size-font-small;
    font-weight:$font-weight-bold;
    margin-bottom: $space-6;

    & > div {
		background: $color-brand-white-darker;
		border-radius: $radii-4;
		color: $color-font-black;
		font-size: $space-10;
		font-weight: $font-weight-semi-bold;
		height: calc(var(--space-16) + 1px);
		margin-left: $space-10;
		min-width: calc(var(--space-16) + 1px);
		padding: $space-4;
		width: auto;
    }
}
.popUp-body__container {
	overflow: hidden !important;
	padding: 0 !important;

	.popUp-body__search {
		display: flex;
		& + ul {
			max-height: 21.25rem;
			overflow: auto;
		}
	}
	.Search {
		&.Search--medium {
			width: 100%;
			.Search__search-icon {
				left : $space-18;
			}
		}
		&__cancel-icon {
			right: $space-20;
		}
		.Search__input {
			border-radius: 0;
			height: 2.938rem;
			padding-left: $space-35;
			padding-right: $space-40;
		}
		& > svg {
			height: $space-14;
			left: $space-16;
			width: $space-14;
		}
	}
	ul {
		padding: $space-18 1.063rem;
		li {
			font-size: $size-font-medium;
			font-weight: $font-weight-medium;
			letter-spacing: -6e-05em;
			line-height: $space-16;
			padding-bottom:$space-40;
			/* Commenting below as we need to display case sensitive values */
			/* text-transform: capitalize; */
			&:first-child {
				padding-top: 0;
			}
			&:last-child {
				padding-bottom: 0;
			}
		}		
	}
}
