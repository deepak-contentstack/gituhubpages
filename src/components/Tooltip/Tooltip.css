@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

@mixin font-Inter {
    font-family: $font-family-primary;
}

.tippy-box[data-animation='fade'][data-state='hidden'] {
    opacity: $opacity-0;
}

.tippy-box[data-inertia][data-state='visible'] {
    transition-timing-function: cubic-bezier(0.54, 1.5, 0.38, 1.11);
}

.tippy-wrapper{
    display: inline-block;
}

.tippy-box-dark {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    
    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-dark-tooltip {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    position: absolute;
    top: -30px;
    left: 0px;
    width:max-content;
    

    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-light {
    background-color: $color-brand-white-base;
    border-radius: $radii-10;
    box-shadow: 0 4px 30px rgba(0, 0, 0, 0.25);
    color: $color-font-base;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: -6e-05em;
    text-align: left;
    @include font-Inter;

    &--primary {
        max-height: 22.81rem;
        max-width: 25rem!important;
        overflow: auto;
        padding: calc(var(--space-16) + 1px) $space-20;
        width: 25rem;

        > .tippy-content {
            .tippy-card {
                border-bottom: 1px solid $color-border-lighter;
                margin-bottom: $space-20;
                padding-bottom: $space-8;
                position: relative;

                .Env-label {
                    color: $color-font-base;
                    font-size: $size-font-small;
                    font-weight: $font-weight-medium;
                    margin-left: $space-5;
                }

                .tippy-card-content {
                    padding: $space-18 0 $space-18 $space-15;

                    .tippy-label {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-bold;
                    }
                    .tippy-value {
                        color:$color-font-tertiary;
                        font-size: $size-font-small;
                        font-weight: $font-weight-medium;
                        letter-spacing: -6e-5em;
                        margin-top: $space-5;
                    }
                    .tippy-lang {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-regular;
                    }
                }
                &:last-child {
                    border-bottom: 0;
                    margin-bottom: 0;
                    padding-bottom: 0;
                }
            }
        }
    }

    &--secondary {
        max-width: 22.5rem !important;
        padding: $space-15 $space-20;
        width: auto;

        >.tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color:$color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }


    &--dynamic {
        padding: $space-15 $space-20;
        > .tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color: $color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }
}

.TooltipWrapper {
    padding:  3.125rem 6.25rem;
}

.KeyShortcut {
    background: $color-brand-primary-base;
    border-radius: $space-5;
    color: $color-font-white;
    font-size: $size-font-tiny;
    font-weight: $font-weight-bold;
    height: $space-20;
    margin-left: $space-10;
    padding: $space-5 0;
    text-align: center;
    width: $space-20;
}

.tippy-box-light-menu {
    border-radius: $radii-6;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
    color: $color-font-base;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    padding: $space-6 $space-10;
    position: relative;
    right: 1px;
    transition: 0ms !important;
    white-space: nowrap;
    .tippy-content {
        transition: 0ms !important;
    }
}

.tippy-arrow-light {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.tippy-arrow-dark {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.closeContainer{
    border-radius: $radii-50;
    cursor: pointer;
    height: $space-32;
    width: $space-32;
    &:hover{
        background: $color-brand-secondary-lighter;
        svg {
            path {
                stroke: $color-base-gray-20;
            }
        }
    }
}
.tippy-content {
    overflow-wrap: break-word;
}