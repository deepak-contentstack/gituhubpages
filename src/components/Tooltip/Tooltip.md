## **Tooltip**

A Tooltip Component can be used to indicate the user what is the functionality of certain element.

### **Tooltip Variant use**

- **dark**: Tooltip with dark variant can be used to show additional information for certain element.

  - props:
    - variant: dark

- **light**: Tooltip that has more details and components as a content (eg: Tooltip Card, Publish status.)

  - props:
    - variant: light

#### **Props**:

- **content**: String or React Component

  - required

- **position**: String

  - required
  - values : right | left | top | bottom

- **variantType**: String

  -optional (defaults to dark)
  -values : dark | light

- **showArrow**: Boolean

  - optional

- **trigger**: String

  - optional
  - default value : 'mouseenter focus'
  - other values: 'click', 'focusin', 'mouseenter click'

* **onHidden**: function
  Invoked once after tooltip hides.
  - optional


* **visible**: Boolean
  - Control the tooltips visibility
  - optional


* **showClose**: Boolean
  - Show a close icon at the top right corner of the tooltip
  - optional

* **onClose**: Functiom
  - Callback function to be executed after close icon is clicked
  - optional


