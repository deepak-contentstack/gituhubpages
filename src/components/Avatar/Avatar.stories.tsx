import React from 'react'
import Avatar, { AvatarInfo } from './Avatar'
import Tooltip from '../Tooltip/Tooltip'

export default {
  title: 'In-progress / Avatar',
  component: Avatar,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/Tn5uYSFQSqblrqrEDt8CcH/Entry-Editor?node-id=603%3A1428',
    },
  },
  argTypes: {
    showPlusSignAfter: {
      control: { type: 'number' },
    },
    showStatus: {
      control: { type: 'boolean' },
      table: {
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: false
        }
      }
    },
    showExtraType: {
      control: { type: 'select', options: ['link', 'button'], },

      table: {
        type: {
          summary: 'Partial'
        }
      }
    },
    type: {
      control: { type: 'select', options: ['text', 'image'] },
      description: "Controls the type of the avatar",
      table: {
        type: {
          summary: 'Partial'
        }
      }
    },
    size: {
      control: {
        type: 'select',
        options: ['small', 'large'],
      },
      table: {
        type: {
          summary: 'Partial'
        }
      }
    },
    showExtraColor: {
      control: {
        type: 'select',
        options: ['light', 'dark']
      },
      table: {
        type: {
          summary: 'Partial'
        }
      }
    }
  },
}

const colorList: any = [
  "#99D8CE",
  "#6BC3FE",
  "#5060C1",
  "#835EC3",
  "#B16DBD",
  "#FF85BC",
  "#FF7E83",
  "#A2D959",
  "#59BA5E"
]

function getRandomColor() {
  const color: any = colorList[Math.floor(Math.random() * colorList.length)];
  return color;
};

const personAvatars = [
  {
    id: 1,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    status: 'active',
    avatarColor: getRandomColor()
  },
  {
    id: 2,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 3,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 4,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 5,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 6,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 7,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 8,
    name: 'Anup Singh',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 9,
    name: 'Amrendra Upadhyay',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
  {
    id: 10,
    name: 'Rutuja Patil',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: getRandomColor(),
    status: 'active'
  },
]
const tableDisable = { table: { disable: true } }
let avatarInfo = <AvatarInfo name="Branko Toshin" email="developer@upwork.com"></AvatarInfo>

export const Default = (args) => <Avatar avatars={[personAvatars[0]]} type={args.type} showExtraType={args.showExtraType} showStatus={args.showStatus} showPlusSignAfter={args.showPlusSignAfter} showExtraColor={args.showExtraColor} testId={args.testId}></Avatar>
Default.args = { type: 'text', showExtraType: 'link', showExtraColor: "light", showPlusSignAfter: 5, showStatus: true };
Default.argTypes = {
  showExtraType: tableDisable,
  showExtraColor: tableDisable,
  showPlusSignAfter: tableDisable,
  avatars: tableDisable,
  size: tableDisable,

}

export const SingleAvatarWithTooltip = (args) => (
  <Tooltip content={avatarInfo} position="right" variantType="light">
    <Avatar avatars={[personAvatars[0]]} type={args.type} size={args.size}></Avatar>
  </Tooltip>
)
SingleAvatarWithTooltip.args = { type: 'text' };
SingleAvatarWithTooltip.argTypes = {
  showExtraType: tableDisable,
  showExtraColor: tableDisable,
  showPlusSignAfter: tableDisable,
  showStatus: tableDisable,
  avatars: tableDisable,
  size: tableDisable
}
export const AvatarList = (args) => (
  <Avatar avatars={personAvatars} type={args.type} showPlusSignAfter={args.showPlusSignAfter} showExtraColor={args.showExtraColor} showExtraType={args.showExtraType} size={args.size} showStatus={args.showStatus}></Avatar>
)
AvatarList.args = { type: 'image', showPlusSignAfter: 5, showExtraType: 'link', showExtraColor: "light", showStatus: true, };
AvatarList.argTypes = {

  showStatus: tableDisable,
  avatars: tableDisable,
}
