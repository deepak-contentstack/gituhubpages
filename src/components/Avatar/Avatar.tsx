import React, { useState } from 'react'
import './Avatar.css'
//import '../DropDown/Dropdown'
import Tooltip from '../Tooltip/Tooltip';
// import cbModal, { ModalFooter, ModalHeader, ModalBody } from '../Modal/Modal'
import Button from '../Button/Button'
// import Search from '../Search/Search'

const defaultProps = {
  avatars: [],
  showPlusSignAfter: 5,
  type: 'text',
  showExtraType: 'button',
  showExtraColor: 'light',
  size: 'small',
  showStatus: false,
  testId: 'cs-avatar'
}

type DefaultProps = Readonly<typeof defaultProps>

export type AvatarProps = {
  avatars?: Array<AvatarData>
  showPlusSignAfter?: number
  type?: 'text' | 'image'
  showExtraType?: string
  showExtraColor?: string
  size?: 'small' | 'large'
  showStatus?: boolean,
  testId?: string
} & Partial<DefaultProps>

export type AvatarData = {
  id: number
  name: string
  image: string
  email: string
  avatarColor: string
}

function displayAvatarContent(type, avatar) {
  let nameInitials = avatar && avatar.name && avatar.name.split(' ')
  let firstInitial = nameInitials[0]
  let lastInitial = nameInitials[nameInitials.length - 1]
  let nameInitialsShort = firstInitial.substring(0, 1) + lastInitial.substring(0, 1)
  if (type === 'image') {
    return <img src={avatar.image} alt={avatar.name} className="Avatar__image" />
  }
  return nameInitialsShort;
}

const searchUser = (users, searchText = '') => {
  if (searchText) {
    return users.filter((user) => {
      return user.name.toLowerCase().includes(searchText.toLowerCase())
    });
  }
  return users;
}
const PopUpComponent = (props) => {
  const { avatars, closeModal } = props;
  const [searchText, setSearchText] = useState('')
  const updateSearchText = (value) => {
    setSearchText(value)
  }
  const filterUsers = searchUser(avatars, searchText)

  return (
    <>

    </>
  )
}

function renderExtraAvatarsButton(showExtraAvatars, showExtraType, extraCount, showExtraColor, avatars) {
  const popUpMembersList = (event: any) => {
    event.stopPropagation();
    event.preventDefault();
    // cbModal({
    //   component: props => <PopUpComponent {...props} avatars={avatars} />,
    //   modalProps: {
    //     size: 'tiny',
    //     style: {
    //       overlay: {
    //         zIndex: '1000',
    //       },
    //       content: {
    //       },
    //     },
    //   },
    // })
  }

  if (showExtraAvatars) {
    if (showExtraType === 'button') {
      return (
        <button
          className={`AvatarList__expand-button${extraCount > 9 ? ' AvatarList__expand-button--tiny-text' : ''
            }`}
          onClick={popUpMembersList}
        >
          +{extraCount}
        </button>
      )
    } else if (showExtraType === 'link') {
      return (
        <a role="link"
          className={`AvatarList__expand-link${showExtraColor === 'dark' ? ' AvatarList__expand-link--dark' : ''
            }`}
          onClick={popUpMembersList}
        >
          +{extraCount} others
        </a>
      )
    }
  }
}

function Avatar({ avatars, showPlusSignAfter, type, showExtraType, showExtraColor, size, showStatus, testId }: AvatarProps) {
  if (avatars.length === 1) {
    let avatar = avatars[0]
    return (
      <Tooltip content={avatar.name || avatar.email} position="bottom" showArrow={false}>
        <div style={{ backgroundColor: avatar.avatarColor }} className="Avatar Avatar--single">
          <span className={`Avatar__link ${!avatar.avatarColor ? 'Avatar__link__grey' : ''}`}>
            {displayAvatarContent(type, avatar)}
          </span>
          {showStatus && avatar.status === 'active' && <span className="Avatar__active"></span>}
        </div>
      </Tooltip>
    )
  } else {
    let showExtraAvatars = avatars.length > showPlusSignAfter ? true : false
    let visibleAvatars = []
    let hiddenAvatars = []
    if (showExtraAvatars) {
      visibleAvatars = avatars.slice(0, showPlusSignAfter)
      hiddenAvatars = avatars.slice(showPlusSignAfter)
    } else {
      visibleAvatars = avatars
    }
    return (
      <div className="AvatarList" data-test-id={testId}>
        {visibleAvatars.map(avatar => {
          return (
            <div key={avatar.id}>
              <Tooltip content={avatar.name || avatar.email} position="bottom" showArrow={false}>
                <div style={{ backgroundColor: avatar.avatarColor }} className={`Avatar Avatar--${size}`}>
                  <span className={`Avatar__link ${!avatar.avatarColor ? 'Avatar__link__grey' : 'Avatar__link__white'}`}>
                    {displayAvatarContent(type, avatar)}
                  </span>
                  {showStatus && avatar.status === 'active' && <span className="Avatar__active"></span>}
                </div>
              </Tooltip>
            </div>
          )
        })}
        {renderExtraAvatarsButton(
          showExtraAvatars,
          showExtraType,
          hiddenAvatars.length,
          showExtraColor,
          avatars
        )}
      </div>
    )
  }
}

const AvatarInfo = ({ name, email }) => {
  return (
    <div className="Avatar__info">
      <h6 className="Avatar__name">{name}</h6>
      <span className="Avatar__email">{email}</span>
    </div>
  )
}

Avatar.defaultProps = defaultProps

export default Avatar
export { AvatarInfo }
