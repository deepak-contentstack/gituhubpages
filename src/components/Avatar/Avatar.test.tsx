import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Avatar, { AvatarInfo } from './Avatar';
import Tooltip from '../Tooltip/Tooltip';
import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

jest.mock('react-dom', () => {
  const original = jest.requireActual('react-dom');
  return {
    ...original,
    createPortal: node => node,
  };
});

describe('Avatar  Component', () => {
  const personAvatars = Array.from({ length: 6 }, (_, i) => ({
    id: i,
    name: 'Branko Toshin',
    image: 'https://via.placeholder.com/32x32/ff4988/ff4988?text=1',
    email: 'fe_developer@gmail.com',
    avatarColor: "#99D8CE",
    status: 'active'
  }));
  it('render Single Avatar ', () => {

    const tree = renderer.create(<Avatar avatars={[personAvatars[0]]} />).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render Single Avatar with image', () => {
    const tree = renderer.create(<Avatar avatars={[personAvatars[0]]} type="image" />).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render Single Avatar with Tolltip ', () => {
    let avatarInfo = <AvatarInfo name="Branko Toshin" email="developer@upwork.com"></AvatarInfo>

    const tree = renderer.create(<Tooltip content={avatarInfo} position="right" variantType="light" showArrow={false}>
      <Avatar avatars={[personAvatars[0]]} type="image"></Avatar>
    </Tooltip>).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render Avatar List ', () => {
    const tree = mount(<Avatar avatars={personAvatars} />);
    expect(tree.find('.Avatar')).toHaveLength(personAvatars.length - 1);
  });
  it('render Avatar List with images ', () => {
    const tree = mount(<Avatar avatars={personAvatars} type="image" />);
    expect(tree.find('.Avatar__image')).toHaveLength(personAvatars.length - 1);
  });

  it('render Avatar List with Link Expander ', () => {
    const tree = mount(<Avatar avatars={personAvatars} type="image" showPlusSignAfter={5} showExtraType="link" />);
    expect(tree.find('.AvatarList__expand-link')).toHaveLength(1);
  });
  it('render Avatar List with Button Expander ', () => {
    const tree = mount(<Avatar avatars={personAvatars} type="image" />);
    expect(tree.find('.AvatarList__expand-button')).toHaveLength(1);
  });
  it('render Single Avatar with Status ', () => {
    const tree = mount(<Avatar avatars={[personAvatars[0]]} showStatus={true} />);
    expect(tree.find('.Avatar__active')).toHaveLength(1);
  });
  it('render Avatar List with Status ', () => {
    const tree = mount(<Avatar avatars={personAvatars} showStatus={true} showPlusSignAfter={5} />)
    expect(tree.find('.Avatar__active')).toHaveLength(personAvatars.length - 1);
  });
  it('render Avatar List with Avatars no more than showPlusSignAfter ', () => {
    const tree = mount(<Avatar avatars={personAvatars} showPlusSignAfter={6} />
    );
    expect(tree.find('.tippy-wrapper').children()).toHaveLength(personAvatars.length);
  });

  it('render Avatar List with Extra avatars ', () => {
    const tree = mount(<Avatar avatars={personAvatars} showPlusSignAfter={5} />
    );
    expect(tree.find('.tippy-wrapper').children()).toHaveLength(personAvatars.length - 1);
  });
  it('simulate click on  Avatar List Button expander ', () => {

    const e = { preventDefault: jest.fn(), stopPropagation: jest.fn() };
    const tree = shallow(<Avatar avatars={personAvatars} showExtraType="button" />);
    tree.find('.AvatarList__expand-button').simulate('click', e);
    expect(e.preventDefault).toHaveBeenCalledTimes(1);

  });

});