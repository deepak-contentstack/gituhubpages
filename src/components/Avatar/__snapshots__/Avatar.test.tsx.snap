// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar  Component render Single Avatar  1`] = `
Array [
  <div
    className="tippy-wrapper"
    data-test-id="cs-tooltip"
  >
    <div
      className="Avatar Avatar--single"
      style={
        Object {
          "backgroundColor": "#99D8CE",
        }
      }
    >
      <span
        className="Avatar__link "
      >
        BT
      </span>
    </div>
  </div>,
  "Branko Toshin",
]
`;

exports[`Avatar  Component render Single Avatar with Tolltip  1`] = `
Array [
  <div
    className="tippy-wrapper"
    data-test-id="cs-tooltip"
  >
    <div
      className="tippy-wrapper"
      data-test-id="cs-tooltip"
    >
      <div
        className="Avatar Avatar--single"
        style={
          Object {
            "backgroundColor": "#99D8CE",
          }
        }
      >
        <span
          className="Avatar__link "
        >
          <img
            alt="Branko Toshin"
            className="Avatar__image"
            src="https://via.placeholder.com/32x32/ff4988/ff4988?text=1"
          />
        </span>
      </div>
    </div>
    Branko Toshin
  </div>,
  <div
    className="Avatar__info"
  >
    <h6
      className="Avatar__name"
    >
      Branko Toshin
    </h6>
    <span
      className="Avatar__email"
    >
      developer@upwork.com
    </span>
  </div>,
]
`;

exports[`Avatar  Component render Single Avatar with image 1`] = `
Array [
  <div
    className="tippy-wrapper"
    data-test-id="cs-tooltip"
  >
    <div
      className="Avatar Avatar--single"
      style={
        Object {
          "backgroundColor": "#99D8CE",
        }
      }
    >
      <span
        className="Avatar__link "
      >
        <img
          alt="Branko Toshin"
          className="Avatar__image"
          src="https://via.placeholder.com/32x32/ff4988/ff4988?text=1"
        />
      </span>
    </div>
  </div>,
  "Branko Toshin",
]
`;
